tests/unit/rankers/test_algorithms.py

```diff
@@ -24,18 +26,93 @@ class MockCorpusItems:
     def get_topics():
         return [business_topic, technology_topic, gaming_topic, health_topic, entertainment_topic]
 
+    @staticmethod
+    def get_publishers():
+        return ["thedude.com", "walter.com", "donnie.com", "bowling.com", "abide.com"]
+
     @staticmethod
     def get_recs():
         topics = MockCorpusItems.get_topics()
         # 5 topics x 3 articles
         recs = []
         for t in topics:
             for j in range(3):
-                recs.append(CorpusItemModel(id=f'{t.name}-rec-{j}', topic=t.corpus_topic_id))
+                recs.append(CorpusItemModel(id=f'{t.name}-rec-{j}', topic=t.corpus_topic_id,
+                                            publisher="that_one_site.com"))
+
+        return recs
+
+    @staticmethod
+    def get_recs_with_publishers():
+        topics = MockCorpusItems.get_topics()
+        publishers = MockCorpusItems.get_publishers()
+        # 5 topics x 5 publishers x 3 articles
+        recs = []
+        for t in topics:
+            for p in publishers:
+                for j in range(3):
+                    recs.append(CorpusItemModel(id=f'{t.name}-{p}-rec-{j}', topic=t.corpus_topic_id,
+                                                publisher=p))
 
         return recs
 
 
+def test_spread_topics():
+    recs = MockCorpusItems.get_recs()
+    topics = set([r.topic for r in recs])
+    reordered = spread_topics(recs)
+
+    for i, r in enumerate(reordered[:15]):
+        if i == 0:
+            prev_topic = r.topic
+        else:  # could write a more complex test for bigger spreads
+            assert r.topic != prev_topic
+            prev_topic = r.topic
+
+    # there are three sets of items per topic
+    c5 = Counter([r.topic for r in reordered[:5]])
+    c10 = Counter([r.topic for r in reordered[:10]])
+    c15 = Counter([r.topic for r in reordered[:15]])
+    for t in topics:
+        assert c5[t] == 1
+        assert c10[t] == 2
+        assert c15[t] == 3
+
+
+
+def test_spread_topics_publishers():
+    recs = MockCorpusItems.get_recs_with_publishers()
+    topics = set([r.topic for r in recs])
+
+    for pub_spread in [2, 3, 4]:
```

Comment:
Suggestion: A parameterized test makes it a little clearer that the test is executed multiple times and removes a for-loop.
```suggestion
@pytest.mark.parametrize("pub_spread", [(2, ), (3, ), (4, )])
def test_spread_topics_publishers(pub_spread):
    recs = MockCorpusItems.get_recs_with_publishers()
    topics = set([r.topic for r in recs])

```